tests/foreman/api/test_computeresource_gce.py

```diff
@@ -0,0 +1,360 @@
+
+        :id: ec7ac618-ed1d-4b36-af7b-995cf4f78341
+
+        :BZ: 1794744
+
+        :steps:
+            1. Create a GCE Compute Resource.
+            2. Add a finish template based image in GCE CR.
+
+        :expectedresults: Finish template image should be added in GCE CR along with username
+
+        :CaseLevel: Integration
+        """
+        assert module_gce_finishimg.compute_resource.id == module_gce_compute.id
+        assert module_gce_finishimg.uuid == gce_latest_rhel_uuid
+        assert module_gce_finishimg.username == finishuser
+
+    @tier3
+    def test_positive_create_cloud_init_image(
+            self, module_gce_cloudimg, module_gce_compute, gce_custom_cloudinit_uuid):
+        """Cloud Init template image along with username is being added in GCE CR
+
+        :id: 5a30c227-324a-49f9-932f-ce80da8611d0
+
+        :steps:
+            1. Create a GCE Compute Resource.
+            2. Add a cloud init template in GCE CR.
+
+        :expectedresults: Cloud init image should be added in GCE CR along with username
+
+        :CaseLevel: Integration
+        """
+        assert module_gce_cloudimg.compute_resource.id == module_gce_compute.id
+        assert module_gce_cloudimg.uuid == gce_custom_cloudinit_uuid
+        assert module_gce_cloudimg.username == clouduser
+
+
+class TestGCEHostProvisioningTestCase:
+    """GCE Host Provisioning Tests
+
+    :Requirement: Host
+
+    :CaseComponent: Hosts
+    """
+    @pytest.fixture(scope='class', autouse=True)
+    def class_setup(self, request, gce_latest_rhel_uuid, gce_domain):
+        """
+        Sets Constants for all the Tests, fixtures which will be later used for assertions
+        """
+        request.cls.mtype = 'g1-small'
+        request.cls.network = 'default'
+        request.cls.volsize = '13'
+        request.cls.hostname = gen_string('alpha')
+
+        request.cls.fullhostname = '{}.{}'.format(self.hostname, gce_domain.name).lower()
+
+        request.cls.compute_attrs = {
+            'image_id': gce_latest_rhel_uuid,
+            'machine_type': self.mtype,
+            'network': self.network,
+            'associate_external_ip': True,
+            'volumes_attributes': {
+                '0': {'size_gb': self.volsize}
+            }
+        }
+
+    @pytest.fixture(scope='class')
+    def class_host(
+            self, googleclient, module_architecture, gce_domain, gce_hostgroup,
+            module_org, module_os, module_location, gce_latest_rhel_uuid, module_gce_finishimg):
+        """Provisions the host on GCE
+
+        Later in tests this host will be used to perform assertions
+        """
+        host = entities.Host(
+            architecture=module_architecture, compute_attributes=self.compute_attrs,
+            domain=gce_domain, hostgroup=gce_hostgroup, organization=module_org,
+            operatingsystem=module_os, location=module_location, name=self.hostname,
+            provision_method='image', image=module_gce_finishimg,
```

Comment:
Do you plan to cover `user data/cloud-init ` host create as well ?

Reply:
The cloud-init has been covered by UI plus the CLI automation would be based on Cloud-init.